src/main/java/randoop/generation/GrtFuzzing.java

```diff
@@ -0,0 +1,480 @@
+   *
+   * <p>This method does not fuzz void, char, boolean, or byte types.
+   *
+   * @param sequence the (non-null) sequence to fuzz
+   * @return a sequence consisting of {@code sequence} with additional fuzzing statements appended
+   *     at the end, and a count of the number of fuzzing statements added. If no fuzzing statements
+   *     are added, the original sequence is returned, and the count is 0.
+   */
+  public static GrtFuzzingAndNumStatements fuzz(Sequence sequence) {
+    // The number of fuzzing statements added to the sequence.
+    AtomicInteger fuzzStatementCount = new AtomicInteger(0);
+
+    Type outputType = sequence.getLastVariable().getType();
+
+    // Do not fuzz void, char, boolean, or byte.
+    if (outputType.isVoid()
+        || outputType.runtimeClassIs(char.class)
+        || outputType.runtimeClassIs(Character.class)
+        || outputType.runtimeClassIs(boolean.class)
+        || outputType.runtimeClassIs(Boolean.class)
+        || outputType.runtimeClassIs(byte.class)
+        || outputType.runtimeClassIs(Byte.class)) {
+      return new GrtFuzzingAndNumStatements(sequence, 0);
+    }
+
+    Class<?> outputClass = outputType.getRuntimeClass();
+    List<Executable> fuzzingOperations = new ArrayList<>();
+
+    // Append input statements for fuzzing operations to the sequence.
+    try {
+      if (outputClass.isPrimitive()) { // fuzzing primitive numbers
+        sequence = appendGaussianSampleSequence(sequence, outputClass);
+        fuzzingOperations = getNumberFuzzingMethods(outputClass);
+      } else if (outputClass == String.class) { // fuzzing String
+        // Randomly select a fuzzing operation for String.
+        StringFuzzingOperation operation =
+            StringFuzzingOperation.values()[
+                Randomness.nextRandomInt(StringFuzzingOperation.values().length)];
+        sequence = appendStringFuzzingInputs(sequence, operation, fuzzStatementCount);
+        if (fuzzStatementCount.get() == 0) { // sequence not fuzzed, return original sequence
+          return new GrtFuzzingAndNumStatements(sequence, 0);
+        }
+        fuzzingOperations = getStringFuzzingMethod(operation);
+      } else {
+        // TODO: Fuzz other objects based on purity analysis
+        //  return the original sequence for now
+        return new GrtFuzzingAndNumStatements(sequence, 0);
+      }
+    } catch (Exception e) { // All other exceptions are unexpected
+      throw new RandoopBug("GRT Fuzzing failed: " + e.getMessage(), e);
+    }
+
+    Sequence output = sequence;
+
+    // TODO: Implement cast to improve short fuzzing output readability
+    //  (e.g. cast to Integer for Integer.sum when fuzzing short can make Integer.valueOf()
+    //  unnecessary).
+
+    // Append fuzzing operation statements to the sequence.
+    for (Executable executable : fuzzingOperations) {
+      output =
+          appendFuzzingOperation(output, executable, getOutputType(executable), fuzzStatementCount);
+    }
```

Comment:
Would it be possible to concatenate just two sequences with one another?  The `for` loop does a lot of intermediate concatenation, which is less efficient.

Reply:
I don't think we should try to concatenate two sequences right now, but we might in the future. If we create a sequence for concatenation from a list of statements, the statements must have non-negative indexes. However, most fuzzing statements have inputs, so the first statement must have a negative index, which invalidates the sequence. The only other way to create a sequence is to repeatedly call createSequence like the current implementation, which is unfortunately not efficient but works.